tools/github-ingest/ingest.py

```
@@ -0,0 +1,416 @@
+    fetch_fn=None,
+) -> Iterator[dict]:
+    """Yield issue dicts (excluding PRs) fetched from the GitHub API.
+
+    ``fetch_fn`` is injectable for testing:
+    signature ``(url: str, token: str | None) -> tuple[bytes, dict]``
+    where the second element is response headers.
+    """
+    if fetch_fn is None:
+        def fetch_fn(url: str, tok: str | None) -> tuple[bytes, dict]:
+            req = _make_request(url, tok)
+            with urllib.request.urlopen(req, timeout=30) as resp:
```

Ruff (0.15.6)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)